kubetest/main.go

```diff
@@ -257,6 +257,12 @@ type deployer interface {
 	GetClusterCreated(gcpProject string) (time.Time, error)
 }
 
+// publisher is implemented by deployers that want to publish status on success
+type publisher interface {
```

Comment:
refactor the k8s --publish logic to use this interface?

Reply:
Not obvious to me that we get a lot from doing so, because it's called from the class that would implement the interface (but see #6457 - I'm not sure who should be doing the publishing).

I did this as an interface so we didn't have to stub-implement Publish() in every deployer, but I could just do that instead if that's simpler.

Reply:
What's the value of the publisher interface if it is only used once?

Reply:
Just that we don't have to add `Publish()` to every `deployer` (and we want to do it after we've checked leaks / called down, so I _think_ we need a new hook point).  But I'm happy to add the Publish() function to the `deployer` interface instead...

Reply:
Sure that's fine